pyaedt/modeler/Primitives3D.py

```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
         new_object_name = self._oeditor.CreateSphere(vArg1, vArg2)
         return self._create_object(new_object_name)
 
+    @aedt_exception_handler
+    def create_torus(self, center, major_radius, minor_radius, axis=None, name=None, material_name=None):
+        """Create a torus.
+
+        Parameters
+        ----------
+        center : list
+            Center point for the torus in a list of ``[x, y, z]`` coordinates.
+        major_radius : float
+           Major radius of the torus.
+        minor_radius : float
+           Minor radius of the torus.
+        axis : str, optional
+            Axis of revolution.
+            The default is ``None``, in which case the Z axis is used.
+        name : str, optional
+            Name of the torus. The default is ``None``, in which case the
+            default name is assigned.
+        material_name : str, optional
+            Name of the material.  The default is ``None``, in which case the
+            default material is assigned. If the material name supplied is
+            invalid, the default material is assigned.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.Object3d.Object3d`
+            3D object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateTorus
+
+        Examples
+        --------
+
+        >>> from pyaedt import Hfss
+        >>> hfss = Hfss()
+        >>> origin = [0,0,0]
+        >>> torus = hfss.modeler.primitives.create_torus(origin, major_radius=1,
+        ...                                             minor_radius=0.5, axis="Z",
+        ...                                             name="mytorus", material_name="copper")
+
+        """
+        assert len(center) == 3, "Center argument must be a valid 3 elements List."
+        assert major_radius > 0 and minor_radius > 0, "Both major and minor radius must be greater than 0."
+        assert minor_radius < major_radius, "Major radius must be greater than minor radius."
```

Comment:
Do not use assertions, rather raise errors. See:
https://medium.com/@jadhavmanoj/python-what-is-raise-and-assert-statement-c3908697bc62

> Python’s assert statement is a debugging aid, not a mechanism for handling run-time errors. The goal of using assertions is to let developers find the likely root cause of a bug more quickly. An assertion error should never be raised unless there’s a bug in your program.

Reply:
Done